DEVGUIDE.md

```diff
@@ -0,0 +1,771 @@
+                success = failure_percentage <= treshold_failure_percentage
+                keyword = {
+                    'name': " | ".join(row),
+                    'pass': success,
+                    'tags': [],
+                    'messages': [],
+                    'teardown': [],
+                    'keywords': [],
+                }                               
+                test_case = {
+                'name': 'Locust test case',
+                'tags': [],
+                'setup': [],
+                'teardown': [],
+                'keywords': [keyword]
+                }
+                test_cases.append(test_case)
+            test_suite = {
+            'name': 'Locust test suite, failure percentage {}'.format(treshold_failure_percentage),
+            'tags': self._tags,
+            'setup': [],
+            'teardown': [],
+            'suites': [],
+            'tests': test_cases,
+            }
+            return test_suite
+```
+
+Let's update the tests to match the current functionality. Let's start by defining new data set in `locustenv/locusthandler/resources/requests.csv`:
+
+```
+"Type","Name","Request Count","Failure Count","Median Response Time","Average Response Time","Min Response Time","Max Response Time","Average Content Size","Requests/s","Failures/s","50%","66%","75%","80%","90%","95%","98%","99%","99.9%","99.99%","99.999%","100%"
+"GET","/",10,0,72,75,66,89,2175,0.26,0.00,73,75,86,87,89,89,89,89,89,89,89,89
+"POST","/",5,5,300,323,288,402,157,0.13,0.13,300,330,330,400,400,400,400,400,400,400,400,400
+"GET","/item",30,3,80,79,67,100,2175,0.63,0.00,81,85,86,86,89,92,100,100,100,100,100,100
+"None","Aggregated",39,5,81,109,66,402,1916,1.03,0.13,81,86,87,89,300,330,400,400,400,400,400,400
+```
+
+now we can update the unit tests in `locusthandler/tests/test_locust.py`:
+
+```
+from unittest import TestCase
+
+from pathlib import Path
+from locusthandler import LocustHandler
+
+class TestLocust(TestCase):
+    
+    def setUp(self):
+        config = {'handler': 'LocustHandler', 'keyword': 'run_locust', 'tags': 'LOCUST'}
+        self.handler = LocustHandler(config)
+        path = Path.cwd() / 'resources/requests.csv'
+        self.test_suite = self.handler.parse_results(path)
+
+    def test_suite_has_four_cases(self):
+        self.assertEqual(len(self.test_suite['tests']),4)
+
+    def test_pass_is_true_when_failure_request_percentage_is_zero(self):
+        self.assertEqual(self.test_suite['tests'][0]['keywords'][0]['pass'], True)
+
+    def test_pass_is_true_when_failure_request_percentage_is_ten(self):
+        self.assertEqual(self.test_suite['tests'][2]['keywords'][0]['pass'], True)
+
+    def test_pass_is_false_when_failure_request_percentage_is_above_ten(self):
+        self.assertEqual(self.test_suite['tests'][1]['keywords'][0]['pass'], False)
+
+    def test_failure_percentage_is_ten_by_default(self):
```

Comment:
`test_failure_percentage_is_ten_by_default` tests that the default value of `_get_treshold_failure_percentage` is 10.
`test_pass_is_true_when_failure_request_is_default_value`  ( Notice new name) tests, that the `pass` value is `True` when the amount of failed requests is exactly on the limit, which happens to be 10.

So there is two test cases because another tests that the default value is correct, and another tests that in edge case the value of `pass` will be `True`

Reply:
But given the `pass` wouldn't be `True` if the percentage is not 10 (or whatever the default value is), the test case also tests this invariant.  So the more specific "this is the default" test case is redundant. 

Generally, having these kinds of more specific redundant tests are only good for documenting important details (to explicitly draw attention to them), but in this case, there is no grand point why it is important that the threshold is 10 -- you say it in the guide yourself that this was just flight of fancy. Therefore, we have a redundant test that underlines a design decision that is of no importance. 

In fact, we explicitly implement two different ways for the user to define the threshold themselves because our guess would be just arbitrary. So therefore "shielding" this design decision doesen't, to me, make much sense.

Reply:
It is true that the default value will be tested anyways in the test case which tests the edge case of the treshold. Removed the `test_failure_percentage_is_ten_by_default`